### ISSUE TITLE
NixOS / kernel 4.19 -- none of the gestures work

### ISSUE THREAD

**Issue by srid** [robotrovsky/Linux-Magic-Trackpad-2-Driver#27]
I run NixOS 18.09 with kernel 4.19. 

I can connect the trackpad over bluetooth and can "basically" use it (navigate and click), but none of the gestures including basic two-finger scroll or double tap work. What could have I done wrong?

Kernel module is loaded:
```
# lsmod | grep magic
hid_magicmouse         16384  0
hid                   131072  7 i2c_hid,hidp,usbhid,hid_generic,hid_logitech_dj,hid_logitech_hidpp,hid_magicmouse
# 
```

Here's the relevant NixOS config:
```nix
serices.xserver = {
  libinput.enable = true;
  multitouch.enable = true;
};
```

**Comment by srid** [comment]
From dmesg I gathered that it is using `hid-generic` for some reason, but trying to workaround in vain ...

```
[root@thebeast:~]# echo 0005:004C:0265.0013 > /sys/bus/hid/drivers/hid-generic/unbind

[root@thebeast:~]# echo 0005:004C:0265.0013 > /sys/bus/hid/drivers/magicmouse/bind
-bash: echo: write error: No such device

[root@thebeast:~]# 
```

**Comment by anbroid** [comment]
@srid The first kernel that will have builtin support for this is 4.20, looks like.
http://lists-archives.com/linux-kernel/29261025-hid-for-4-20.html
https://github.com/torvalds/linux/commit/9d7b18668956c411a422d04c712994c5fdb23a4b

**Comment by mwyborski** [comment]
@srid @anbroid is right, the driver doesn't come with 4.19. It will be included in 4.20.
